src/module/item/spell/index.ts

```diff
@@ -231,22 +237,57 @@ class SpellPF2e extends ItemPF2e {
      * This handles heightening as well as alternative cast modes of spells.
      * If there's nothing to apply, returns null.
      */
-    loadVariant(castLevel: number): SpellPF2e | null {
+    loadVariant(options: { castLevel?: number; overlayIds?: string[] } = {}): Embedded<SpellPF2e> | null {
         if (this.original) {
-            return this.original.loadVariant(castLevel);
+            return this.original.loadVariant(options);
         }
+        const { castLevel, overlayIds } = options;
+
+        const override = (() => {
+            // Retrieve and apply variant overlays to override data
+            const heightenEntries = this.getHeightenLayers(castLevel);
+            if (heightenEntries.length === 0 && !overlayIds) return null;
+            let source = this.toObject();
+
+            if (overlayIds) {
+                for (const overlayId of overlayIds) {
+                    const overlayData = this.data.data.overlay?.[overlayId];
+                    if (!overlayData) {
+                        throw ErrorPF2e(
+                            `Spell ${this.name} (${this.uuid}) does not have an overlay with id: ${overlayId}`
+                        );
+                    }
+                    switch (overlayData.overlayType) {
+                        case "override": {
+                            // Sanitize data
+                            delete source.data.overlay;
+                            source.data.rules = [];
+
+                            source = mergeObject(source, overlayData, { overwrite: true });
+                            break;
+                        }
+                    }
+                    // Store overlay type as a flag on the variant spell
+                    source.flags = mergeObject(source.flags, { pf2e: { overlayType: overlayData.overlayType } });
```

Comment:
Not sure if this is the right call. If there is more than one overlay spell, the one prioritized is the one applied second. Is there a reason to store in the flags?